.gitlab-ci.yml

```diff
@@ -0,0 +1,212 @@
+stages:
+- runall
+- build
+- collate
+
+# This is just a config to help trigger rest of the builds
+run_all_builds:
+  image: ubuntu:latest
+  stage: runall
+  variables:
+    GIT_STRATEGY: none
+  script:
+  - pwd
+  when: manual
+  allow_failure: false
+
+# Build
+build_osx_runtime:
+  stage: build
+  tags:
+    - buildfarm
+    - darwin
+  script:
+  - git submodule update --init --recursive
+  - echo -e "[auth]\nono.schemes=https\nono.prefix=ono.unity3d.com\nono.username=$GITLAB_UNITY_META_ONO_USER\nono.password=$GITLAB_UNITY_META_ONO_PASS\n" >> ~/.hgrc
+  - chmod +x external/buildscripts/bee
+  - cd external/buildscripts
+  - ./bee
+  - cd ../..
+  - perl external/buildscripts/build_runtime_osx.pl
+  - mkdir -p incomingbuilds/osx-i386
+  - cp -r builds/ incomingbuilds/osx-i386/
+  artifacts:
+    paths:
+    - incomingbuilds/osx-i386
+    expire_in: 1 week
+# Important! Do not remove this after_script!!  
+  after_script:
+    - /opt/post_build_script.sh
+
+build_osx_classlibs:
+  stage: build
+  tags:
+    - buildfarm
+    - darwin
+  script:
+  - git submodule update --init --recursive
+  - echo -e "[auth]\nono.schemes=https\nono.prefix=ono.unity3d.com\nono.username=$GITLAB_UNITY_META_ONO_USER\nono.password=$GITLAB_UNITY_META_ONO_PASS\n" >> ~/.hgrc
+  - chmod +x external/buildscripts/bee
+  - cd external/buildscripts
+  - ./bee
+  - cd ../..
+  - perl external/buildscripts/build_classlibs_osx.pl
+  - mkdir -p incomingbuilds/classlibs
+  - cp -r ZippedClasslibs.tar.gz incomingbuilds/classlibs/
+  artifacts:
+    paths:
+    - incomingbuilds/classlibs
+    expire_in: 1 week
+# Important! Do not remove this after_script!!  
+  after_script:
+    - /opt/post_build_script.sh
+
+build_android:
+  stage: build
+  tags:
+    - buildfarm
```

Comment:
I started with build farm runners since we needed to access Ono. We might not need to use them going forward.

Reply:
I will change to use bokken in next PR